packages/ui-text-input/src/TextInput/README.md

```diff
@@ -301,6 +298,47 @@ Focusable content will be focused separately from the input itself.
   render(<ExtraContentExample />)
   ```
 
+Another common usecase is to add an `IconButton` at the end of a TextInput, e.g. for revealing the content of a password field. In these cases, please use the `withBorder={false}` and `withBackground={false}` props for the IconButton.
+
+```js
+---
+type: example
+---
+const InputsWithButtonsExample = () => {
+  const [passwordValue, setPasswordValue] = useState('')
+  const [showPassword, setShowPassword] = useState(false)
+  return (
+    <TextInput
+      renderLabel="Password"
+      type={showPassword ? 'text' : 'password'}
+      placeholder="Find something..."
+      value={passwordValue}
+      onChange={(e, newValue) => setPasswordValue(newValue)}
+      renderAfterInput={
+        <IconButton withBorder={false} withBackground={false} onClick={() => setShowPassword(prevState => !prevState)}>
+          {showPassword ? <IconOffLine/> : <IconEyeLine/>}
+        </IconButton>
+      }
+    />
+  )
+}
+render(<InputsWithButtonsExample />)
+```
+
+Having a clear button for a search field is also common but you don't need to implement anything to make that work, only set the `type` of the input to `search`:
+
+```js
+---
+type: example
+---
+<TextInput
+  renderLabel="Search"
+  type="search"
+  placeholder="Find something..."
+  renderBeforeInput={<IconSearchSolid/>}
+/>
+```
```

Comment:
isnt it an a11y problem that the close button here does not have a focus ring and is not in the keyboard tab order?